xdsl/transforms/experimental/convert_stencil_to_ll_mlir.py

```diff
@@ -649,7 +650,7 @@ def return_target_analysis(module: builtin.ModuleOp):
 
 
 class StencilTypeConversion(TypeConversionPattern):
-    @attr_type_rewrite_pattern
+    @attr_constr_rewrite_pattern(StencilTypeConstr)
```

Comment:
Is this a different change?

Reply:
I guess it's more related to the previous change, let me revert this and see what happens

Reply:
Nope it's related to this change, since we use the attr_def instead of the type hints to construct the constraint, we can't do it on the ABC that is StencilTypeConstr since it's not really an attribute. We have to construct the constraint manually instead.